Admin endpoints

kordano opened the pull request.

kordano committed:
integrate connection into middleware

kordano committed:
add database connection middleware

kordano committed:
adjust to new datahike version, add connection list

kordano committed:
add description

kordano committed:
cleanup connections after shutdown

kordano committed:
use latests Datahike version

grischoun committed:
Adds localhost:4000 as a valid cross-origin as it is used by datahike-frontend.

kordano committed:
add query administration

kordano committed:
Merge branch 'admin-endpoints' of github.com:replikativ/datahike-server into admin-endpoints

kordano committed:
create and connect to dbs on creation

kordano committed:
add db handler

kordano committed:
add db endpoint

kordano committed:
add better db representation

kordano committed:
Merge branch 'master' into admin-endpoints

kordano committed:
add db

TimoKramer commented on src/clj/datahike_server/server.clj:
```diff
@@ -52,57 +65,100 @@
+               :parameters {:body ::datoms-request :header ::db-header}
+               :handler    h/datoms}}]
 
    ["/seek datoms"
```
is there a space on purpose? it seems a typo to me.

TimoKramer committed:
Add basic integration tests

TimoKramer committed:
Format files with cljfmt

- Add corrected formatting for all clj files
- Add CI step to check formatting
- CI fails on bad formatting

TimoKramer commented:
there is some room for improvements but not necessarily in scope of this PR:

- slashes at the end of the url don't work
- config via environment does not work with multiple databases (we spoke about it)
- does it make sense to create and delete databases on a running server? delete the files?
- should the db-name be part of the url? this might be more resty
- there should be examples in the swagger doc
- should q have limit and offset params?

kordano committed:
Merge pull request #14 from TimoKramer/integration-tests

added first round of integration tests

kordano committed:
Merge pull request #15 from TimoKramer/formatting

Formatting
